src/suricata.h

```diff
@@ -106,12 +106,16 @@ enum {
 enum EngineMode {
     ENGINE_MODE_UNKNOWN,
     ENGINE_MODE_IDS,
+    /* order matters, we need to be able to do IPS is true for >= ENGINE_MODE_IPS */
     ENGINE_MODE_IPS,
+    ENGINE_MODE_FIREWALL,
 };
 
+void EngineModeSetFirewall(void);
 void EngineModeSetIPS(void);
 void EngineModeSetIDS(void);
 int EngineModeIsUnknown(void);
+int EngineModeIsFirewall(void);
```

Comment:
Maybe a little picky.. But are we fireall mode OR IPS mode?  Or are we IPS mode with firewall enabled/disabled? Or is firewall mode considered a superset of IPS mode? I think its worth trying to clarify that somewhere.